substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ReachabilityHandlerFeature.java

```diff
@@ -52,8 +54,9 @@
 @AutomaticallyRegisteredFeature
 public class ReachabilityHandlerFeature extends ReachabilityHandler implements InternalFeature {
 
-    private final IdentityHashMap<Object, Set<Object>> activeHandlers = new IdentityHashMap<>();
-    private final IdentityHashMap<Object, Map<Object, Set<Object>>> triggeredHandlers = new IdentityHashMap<>();
+    private static final Map<Object, Set<Object>> NULL_MAP = new IdentityHashMap<>();
+    private final ConcurrentIdentityHashMap<Object, ConcurrentHashMap.KeySetView<Object, Boolean>> activeHandlers = new ConcurrentIdentityHashMap<>();
```

Comment:
Why change the generic type from `Set` to `ConcurrentHashMap.KeySetView`? The field can just be defined as `Map<Object, Set<Object>> activeHandlers`.

Same for all the other type definitions.

Reply:
I guess I just like explicit types :)

Changed.